### ISSUE TITLE
UltraVNC viewer do not pass Cyrillic letter "у" to the server

### ISSUE THREAD

**Issue by mgorchak-blackberry** [ultravnc/UltraVNC#211]
Hi,

UltraVNC viewer do not pass following letter:

'у', XK_Cyrillic_u, hex:0x6d5, dec:1749.

from client to the server, server receives just simple slash '/' character. The rest Ukrainian and most Cyrillic letters what I can test were passed properly.

Thanks!

**Comment by mark99i** [comment]
> Hi,
> 
> UltraVNC viewer do not pass following letter:
> 
> 'у', XK_Cyrillic_u, hex:0x6d5, dec:1749.
> 
> from client to the server, server receives just simple slash '/' character. The rest Ukrainian and most Cyrillic letters what I can test were passed properly.
> 
> Thanks!
> 

this is not going to be fixed, use TightVNC

**Comment by jose94752** [comment]
Same problem as the issue #214 using a variable is a std::string in Clipboard.h !!! Not a std::wstring or std::u16string or std::u32string. The code copy extended characters in a string.
Like I copy into notepad ANSI.txt file these
Test Cyrillic Character
'?', XK_Cyrillic_u, hex:0x6d5, dec:1749.

**Comment by Neustradamus** [comment]
Character problem is linked to tickets:
- https://github.com/ultravnc/UltraVNC/issues/332
- https://github.com/ultravnc/UltraVNC/issues/323
- https://github.com/ultravnc/UltraVNC/issues/253
- https://github.com/ultravnc/UltraVNC/issues/240
- https://github.com/ultravnc/UltraVNC/issues/214
- https://github.com/ultravnc/UltraVNC/issues/211
- https://github.com/ultravnc/UltraVNC/issues/124
- https://github.com/ultravnc/UltraVNC/issues/72
- https://github.com/ultravnc/UltraVNC/issues/64
- https://github.com/ultravnc/UltraVNC/issues/7#issuecomment-1003742782

For me it is linked to keymap.

**Comment by RudiDeVos** [comment]
keyboard and copy/paste support unicode UltraVNC<-> UltraVNC  
version 1.7.1.5

**Comment by Neustradamus** [comment]
@mgorchak-blackberry, @mark99i, @jose94752: At this time, can you try the 1.7.1.9-dev release build?
- https://uvnc.eu/download/1710/UltraVNC_1719-dev.zip
- https://uvnc.eu/download/1710/UltraVNC_1719_x64_Setup.exe
- https://uvnc.eu/download/1710/UltraVNC_1719_x86_Setup.exe

**Comment by sniak** [comment]
@Neustradamus
Hi, 
The Cyrillic letter "у" still doesn't work in build 1719.

**Comment by sniak** [comment]
@Neustradamus 
Hi, 
Same problem 
I am using a client on windows 11 and a standard server on macos (maybe this information will help)

**Comment by sniak** [comment]
@Neustradamus the letter "y" still doesn't work

**Comment by sniak** [comment]
@Neustradamus Hi, the problem persists

**Comment by Neustradamus** [comment]
@sniak: Have you installer UltraVNC in 2 sides (Server and Viewer)?

@RudiDeVos: This ticket must be reopened...

**Comment by sniak** [comment]
@Neustradamus Is this necessary for testing now? I didn't install a server; I used a standard MacOS server, which I connected to through a client.
It turns out it's impossible to test on a Mac. I'll test on a Windows host later and report back in a day or two.

**Comment by Neustradamus** [comment]
@sniak: Ah, sorry, I have forgotten that you have specified that you have a mac!
It is integrated macOS VNC Server?
So I suppose that there is a problem in MacOS VNC Server.
What is your macOS version?

I do not know but have you possibility to enable logging in the macOS VNC Server?

I hope that @RudiDeVos can add log options in UltraVNC Viewer, it will be better to debug in all cases.

If you can try the UltraVNC Server on a Windows machine in parallel to have UltraVNC 1.8.0.5 in 2 sides, it will be nice.
Of course, it will be not solved the problem with macOS :/

**Comment by sniak** [comment]
@Neustradamus 
Hi. I was able to test the client and server on two different Windows machines. The "у" issue persisted.
Both machines are running Windows 11.

The Mac is running MacOS 26.3.1.

**Comment by RudiDeVos** [comment]
Also fixed in 1.8.0.7, same  cause as the geo

**Comment by Neustradamus** [comment]
@RudiDeVos: Good news, thanks!

**Comment by Neustradamus** [comment]
@RudiDeVos: Good work, thanks!

@sniak, I will inform you when 1.8.0.7 will be here!

**Comment by Neustradamus** [comment]
@sniak: At this time, can you try the 1.8.0.7 release build?
- https://uvnc.eu/download/1800/UltraVNC_1807_x64_Setup.exe
- https://uvnc.eu/download/1800/UltraVNC_1807_x86_Setup.exe
- https://uvnc.eu/download/1800/UltraVNC_1807.zip

**Comment by sniak** [comment]
@Neustradamus Hi. I tested build 1.8.0.7 for connecting on MacOS and Windows. It works!

**Comment by Neustradamus** [comment]
@sniak: Nice, happy to read it, thanks for your very fast test between Windows and macOS!

Can you try between 2 Windows OS too?

Hope that other users will try it.

**Comment by sniak** [comment]
@Neustradamus Yes, I just tested the mac-win and win-win combination - both options work.

**Comment by Neustradamus** [comment]
@sniak: Perfect, thanks for your tests/feedback!
We can thank @RudiDeVos for this work!

**Comment by RudiDeVos** [comment]
Thanks all for patience and time to test.
missed the critical spot where this key was excluded

**Comment by RudiDeVos** [comment]
This may already be resolved in the current development branch. The keyboard input pipeline has been updated to use Unicode keysym encoding with `SendInput`/`KEYEVENTF_UNICODE`, which should correctly handle all Unicode characters regardless of the server's active keyboard layout. Could you please test with the next release build and let us know if the issue persists?

Testing Automated answers based on git development :)

**Comment by Neustradamus** [comment]
@mgorchak-blackberry, @mark99i, @jose94752, @sniak: At this time, can you try the 1.8.0.8 release build?
- https://uvnc.eu/download/1800/UltraVNC_1808_x64_Setup.exe
- https://uvnc.eu/download/1800/UltraVNC_1808_x86_Setup.exe
- https://uvnc.eu/download/1800/UltraVNC_1808.zip

**Comment by sniak** [comment]
@Neustradamus @RudiDeVos 
I tested build 1.8.0.8.
Cyrillic works correctly in both Win-Mac and Win-Win configurations.

**Comment by Neustradamus** [comment]
@sniak: Nice, so this ticket is now solved!
Thanks for your tests!

**Comment by RudiDeVos** [comment]
Thanks all for patience the feedback.
